doc/source/placement-group.rst

```diff
@@ -128,77 +171,196 @@ Let's create a placement group. Recall that each bundle is a collection of resou
+
+        // Before you run this code, you need to make sure there are only two "CPU" are available.
+        public static class Counter {
+          public static String ping() {
+            return "pong";
+          }
+        }
+
+        // Construct a list of bundles.
+        List<Map<String, Double>> bundles = new ArrayList<>();
+        Map<String, Double> bundle = new HashMap<>();
+        bundle.put("CPU", 2.0);
+        bundles.add(bundle);
+
+        // Create a placement group and make sure its creation is successful.
+        PlacementGroupCreationOptions options =
+          new PlacementGroupCreationOptions.Builder()
+            .setBundles(bundles)
+            .setStrategy(PlacementStrategy.STRICT_SPREAD)
+            .build();
+
+        PlacementGroup pg = PlacementGroups.createPlacementGroup(options);
+        boolean isCreated = pg.wait(60);
+        Assert.assertTrue(isCreated);
+
+        // Won't be scheduled because there are no 2 cpus.
+        Ray.task(Counter::ping)
+          .setResource("CPU", 1.0)
+          .remote();
+
+        // Will be scheduled because 2 cpus are reserved by the placement group.
+        Ray.task(Counter::ping)
+          .setPlacementGroup(pg, 0)
+          .setResource("CPU", 2.0)
+          .remote();
 
 .. note::
 
   When using placement groups, users should ensure that their placement groups are ready (by calling ``ray.get(pg.ready())``)
   and have the proper resources. Ray assumes that the placement group will be properly created and does *not*
   print a warning about infeasible tasks.
 
-  .. code-block:: python
+  .. tabs::
+    .. group-tab:: Python
+
+      .. code-block:: python
+
+        gpu_bundle = {"GPU": 2}
+        extra_resource_bundle = {"extra_resource": 2}
 
-    gpu_bundle = {"GPU": 2}
-    extra_resource_bundle = {"extra_resource": 2}
+        # Reserve bundles with strict pack strategy.
+        # It means Ray will reserve 2 "GPU" and 2 "extra_resource" on the same node (strict pack) within a Ray cluster.
+        # Using this placement group for scheduling actors or tasks will guarantee that they will
+        # be colocated on the same node.
+        pg = placement_group([gpu_bundle, extra_resource_bundle], strategy="STRICT_PACK")
 
-    # Reserve bundles with strict pack strategy.
-    # It means Ray will reserve 2 "GPU" and 2 "extra_resource" on the same node (strict pack) within a Ray cluster.
-    # Using this placement group for scheduling actors or tasks will guarantee that they will
-    # be colocated on the same node.
-    pg = placement_group([gpu_bundle, extra_resource_bundle], strategy="STRICT_PACK")
+        # Wait until placement group is created.
+        ray.get(pg.ready())
 
-    # Wait until placement group is created.
-    ray.get(pg.ready())
+    .. group-tab:: Java
+
+      .. code-block:: java
+
+        // Before you run this code, you need to make sure there are two "CPU" and two "extra_resource" are available.
```

Comment:
we can also init a cluster with these resources in Java
set system property ray.head-args

Reply:
updated.